src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -656,7 +648,11 @@ private function restoreResponseBody(Request $request, Response $response)
                 $response->headers->set('Content-Length', strlen($response->getContent()));
             }
         } elseif ($response->headers->has('X-Body-File')) {
-            $response->setContent(file_get_contents($response->headers->get('X-Body-File')));
+            // Response does not include possibly dynamic content (ESI, SSI), so we need
+            // not handle the content for HEAD requests
+            if (!$request->isMethod('HEAD')) {
+                $response->setContent(file_get_contents($response->headers->get('X-Body-File')));
```

Comment:
This tries to keep part of the performance optimization removed above – if `X-Body-File` is intended to mark responses that do not contain dynamic content like ESI/SSI...?